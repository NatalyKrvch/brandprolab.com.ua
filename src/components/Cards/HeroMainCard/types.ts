export interface HeroMainCardProps {
  title: string;
  subtitle: string;
  description: string;
  buttonText: string;
  buttonLink: string;
  iconURL: string;
  hasBackground?: boolean;
}
