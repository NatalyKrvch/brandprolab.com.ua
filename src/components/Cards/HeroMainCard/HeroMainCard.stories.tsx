import type { Meta, StoryObj } from '@storybook/react';

import HeroMainCard from './HeroMainCard';

const meta: Meta<typeof HeroMainCard> = {
  title: 'Cards/HeroMainCard',
  component: HeroMainCard,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof HeroMainCard>;

export const Primary: Story = {
  args: {
    title: 'Відкрий нові можливості',
    subtitle: 'з нашим сервісом',
    description:
      'Це головна картка з яскравим заголовком, підзаголовком і кнопкою дії.',
    buttonText: 'Спробувати зараз',
    buttonLink: '/try-now',
    iconURL: '/icons/rocket.svg',
  },
};
