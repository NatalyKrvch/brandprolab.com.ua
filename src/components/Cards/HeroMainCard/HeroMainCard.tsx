import Image from 'next/image';

import { Button } from '@/components/Button';
import { ExternalLink } from '@/components/Links';

import { SimpleCard } from '../SimpleCard';
import { HeroMainCardProps } from './types';

const HeroMainCard = ({
  title,
  subtitle,
  description,
  buttonText,
  buttonLink,
  iconURL,
}: HeroMainCardProps) => {
  return (
    <SimpleCard
      className="pt-130 text-white"
      backgroundImageUrl="./images/backgrounds/hero-maincard-bg.svg"
    >
      <Image
        className="mb-6 ml-11"
        width={84}
        height={84}
        src={iconURL}
        aria-hidden="true"
        alt=""
      />

      <h1 className="px-54 text-54 font-bold leading-54 tracking-tight text-white">
        <span className="mb-4 block">{title}</span>

        <span className="block text-right text-40 font-normal leading-40 tracking-normal">
          {subtitle}
        </span>
      </h1>

      <div className="flex flex-col gap-10 px-11 pb-160 pt-11">
        <strong className="max-w-408 px-2 text-xl font-normal">
          {description}
        </strong>

        <ExternalLink
          href={buttonLink}
          aria-label={`Перейти, щоб ${buttonText}`}
        >
          <Button variant="white">{buttonText}</Button>
        </ExternalLink>
      </div>
    </SimpleCard>
  );
};

export default HeroMainCard;
