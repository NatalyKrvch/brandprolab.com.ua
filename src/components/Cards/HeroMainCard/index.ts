export { default as HeroMainCard } from './HeroMainCard';
export * from './types';
