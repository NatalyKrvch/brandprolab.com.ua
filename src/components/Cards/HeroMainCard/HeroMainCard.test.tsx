import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import {
  EXTERNAL_LINK_TEST_ID,
  HERO_MAIN_CARD_ICON_TEST_ID,
  SIMPLE_CARD_TEST_ID,
} from '@/lib/testIDs';

import HeroMainCard from './HeroMainCard';

jest.mock('next/image', () => {
  const MockedImage = ({
    src,
    alt,
    className,
  }: {
    src: string;
    alt: string;
    className?: string;
  }) => (
    <img
      src={src}
      alt={alt}
      className={className}
      data-testid={HERO_MAIN_CARD_ICON_TEST_ID}
    />
  );
  MockedImage.displayName = 'MockedNextImage';
  return MockedImage;
});

describe('HeroMainCard component', () => {
  const defaultProps = {
    title: 'Test Title',
    subtitle: 'Test Subtitle',
    description: 'This is the main card description.',
    buttonText: 'Click Me',
    buttonLink: '/test-link',
    iconURL: '/test-icon.svg',
  };

  it('renders the component correctly', () => {
    const { asFragment } = render(<HeroMainCard {...defaultProps} />);
    const component = screen.getByTestId(SIMPLE_CARD_TEST_ID);

    expect(component).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders icon image with correct src', () => {
    const { asFragment } = render(<HeroMainCard {...defaultProps} />);
    const image = screen.getByTestId(HERO_MAIN_CARD_ICON_TEST_ID);

    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', defaultProps.iconURL);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders title and subtitle', () => {
    const { asFragment } = render(<HeroMainCard {...defaultProps} />);

    expect(screen.getByText(defaultProps.title)).toBeInTheDocument();
    expect(screen.getByText(defaultProps.subtitle)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders description', () => {
    const { asFragment } = render(<HeroMainCard {...defaultProps} />);

    expect(screen.getByText(defaultProps.description)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders button with correct text and link', () => {
    const { asFragment } = render(<HeroMainCard {...defaultProps} />);
    const link = screen.getByTestId(EXTERNAL_LINK_TEST_ID);

    expect(link).toHaveAttribute('href', defaultProps.buttonLink);
    expect(screen.getByText(defaultProps.buttonText)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
