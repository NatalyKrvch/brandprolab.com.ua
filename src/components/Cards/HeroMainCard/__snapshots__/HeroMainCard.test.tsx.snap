// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroMainCard component renders button with correct text and link 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center bg-white rounded-24 tablet:rounded-32 desktop:rounded-40 pt-130 text-white"
    data-testid="simple-card-test-id"
    style="background-image: url(./images/backgrounds/hero-maincard-bg.svg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <img
      alt=""
      class="mb-6 ml-11"
      data-testid="hero-main-card-icon-test-id"
      src="/test-icon.svg"
    />
    <h1
      class="px-54 text-54 font-bold leading-54 tracking-tight text-white"
    >
      <span
        class="mb-4 block"
      >
        Test Title
      </span>
      <span
        class="block text-right text-40 font-normal leading-40 tracking-normal"
      >
        Test Subtitle
      </span>
    </h1>
    <div
      class="pb-160 flex flex-col gap-10 px-11 pt-11"
    >
      <strong
        class="max-w-408 px-2 text-xl font-normal"
      >
        This is the main card description.
      </strong>
      <a
        data-testid="external-link-test-id"
        href="/test-link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="w-full rounded-full py-3 text-xl font-medium leading-26 tracking-wide outline-none transition-all duration-300 hover:outline-offset-0 tablet:py-5 tablet:text-22 desktop:py-6 desktop:text-2xl desktop:leading-6 bg-white font-normal text-blue-default hover:outline-8 hover:outline-whiteOpacity active:bg-teal-light"
          data-testid="button-test-id"
        >
          Click Me
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HeroMainCard component renders description 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center bg-white rounded-24 tablet:rounded-32 desktop:rounded-40 pt-130 text-white"
    data-testid="simple-card-test-id"
    style="background-image: url(./images/backgrounds/hero-maincard-bg.svg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <img
      alt=""
      class="mb-6 ml-11"
      data-testid="hero-main-card-icon-test-id"
      src="/test-icon.svg"
    />
    <h1
      class="px-54 text-54 font-bold leading-54 tracking-tight text-white"
    >
      <span
        class="mb-4 block"
      >
        Test Title
      </span>
      <span
        class="block text-right text-40 font-normal leading-40 tracking-normal"
      >
        Test Subtitle
      </span>
    </h1>
    <div
      class="pb-160 flex flex-col gap-10 px-11 pt-11"
    >
      <strong
        class="max-w-408 px-2 text-xl font-normal"
      >
        This is the main card description.
      </strong>
      <a
        data-testid="external-link-test-id"
        href="/test-link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="w-full rounded-full py-3 text-xl font-medium leading-26 tracking-wide outline-none transition-all duration-300 hover:outline-offset-0 tablet:py-5 tablet:text-22 desktop:py-6 desktop:text-2xl desktop:leading-6 bg-white font-normal text-blue-default hover:outline-8 hover:outline-whiteOpacity active:bg-teal-light"
          data-testid="button-test-id"
        >
          Click Me
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HeroMainCard component renders icon image with correct src 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center bg-white rounded-24 tablet:rounded-32 desktop:rounded-40 pt-130 text-white"
    data-testid="simple-card-test-id"
    style="background-image: url(./images/backgrounds/hero-maincard-bg.svg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <img
      alt=""
      class="mb-6 ml-11"
      data-testid="hero-main-card-icon-test-id"
      src="/test-icon.svg"
    />
    <h1
      class="px-54 text-54 font-bold leading-54 tracking-tight text-white"
    >
      <span
        class="mb-4 block"
      >
        Test Title
      </span>
      <span
        class="block text-right text-40 font-normal leading-40 tracking-normal"
      >
        Test Subtitle
      </span>
    </h1>
    <div
      class="pb-160 flex flex-col gap-10 px-11 pt-11"
    >
      <strong
        class="max-w-408 px-2 text-xl font-normal"
      >
        This is the main card description.
      </strong>
      <a
        data-testid="external-link-test-id"
        href="/test-link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="w-full rounded-full py-3 text-xl font-medium leading-26 tracking-wide outline-none transition-all duration-300 hover:outline-offset-0 tablet:py-5 tablet:text-22 desktop:py-6 desktop:text-2xl desktop:leading-6 bg-white font-normal text-blue-default hover:outline-8 hover:outline-whiteOpacity active:bg-teal-light"
          data-testid="button-test-id"
        >
          Click Me
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HeroMainCard component renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center bg-white rounded-24 tablet:rounded-32 desktop:rounded-40 pt-130 text-white"
    data-testid="simple-card-test-id"
    style="background-image: url(./images/backgrounds/hero-maincard-bg.svg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <img
      alt=""
      class="mb-6 ml-11"
      data-testid="hero-main-card-icon-test-id"
      src="/test-icon.svg"
    />
    <h1
      class="px-54 text-54 font-bold leading-54 tracking-tight text-white"
    >
      <span
        class="mb-4 block"
      >
        Test Title
      </span>
      <span
        class="block text-right text-40 font-normal leading-40 tracking-normal"
      >
        Test Subtitle
      </span>
    </h1>
    <div
      class="pb-160 flex flex-col gap-10 px-11 pt-11"
    >
      <strong
        class="max-w-408 px-2 text-xl font-normal"
      >
        This is the main card description.
      </strong>
      <a
        data-testid="external-link-test-id"
        href="/test-link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="w-full rounded-full py-3 text-xl font-medium leading-26 tracking-wide outline-none transition-all duration-300 hover:outline-offset-0 tablet:py-5 tablet:text-22 desktop:py-6 desktop:text-2xl desktop:leading-6 bg-white font-normal text-blue-default hover:outline-8 hover:outline-whiteOpacity active:bg-teal-light"
          data-testid="button-test-id"
        >
          Click Me
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HeroMainCard component renders title and subtitle 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center bg-white rounded-24 tablet:rounded-32 desktop:rounded-40 pt-130 text-white"
    data-testid="simple-card-test-id"
    style="background-image: url(./images/backgrounds/hero-maincard-bg.svg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <img
      alt=""
      class="mb-6 ml-11"
      data-testid="hero-main-card-icon-test-id"
      src="/test-icon.svg"
    />
    <h1
      class="px-54 text-54 font-bold leading-54 tracking-tight text-white"
    >
      <span
        class="mb-4 block"
      >
        Test Title
      </span>
      <span
        class="block text-right text-40 font-normal leading-40 tracking-normal"
      >
        Test Subtitle
      </span>
    </h1>
    <div
      class="pb-160 flex flex-col gap-10 px-11 pt-11"
    >
      <strong
        class="max-w-408 px-2 text-xl font-normal"
      >
        This is the main card description.
      </strong>
      <a
        data-testid="external-link-test-id"
        href="/test-link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="w-full rounded-full py-3 text-xl font-medium leading-26 tracking-wide outline-none transition-all duration-300 hover:outline-offset-0 tablet:py-5 tablet:text-22 desktop:py-6 desktop:text-2xl desktop:leading-6 bg-white font-normal text-blue-default hover:outline-8 hover:outline-whiteOpacity active:bg-teal-light"
          data-testid="button-test-id"
        >
          Click Me
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
